src/main/java/dk/sdu/se_f22/productmodule/management/ProductManager.java

```diff
@@ -0,0 +1,356 @@
+            }
+        }
+
+        updateSource();
+
+        return succes;
+    }
+
+    @Override
+    public boolean update(String productId, Product p) {
+
+        //This function updates the attributes of a product based on its UUID
+        //By replacing it entirely with a new one. Make sure this new product has all it's attributes set correctly.
+        //Any missing attribute on this other product is also removed from the source file.
+
+        checkForUpdates();
+
+        boolean success = false;
+
+        for(Product pT : productArray){
+            if(pT.get(ProductAttribute.UUID).equalsIgnoreCase(productId)){
+                pT = p;
+                success = true;
+                break;
+            }
+        }
+
+        updateSource();
+
+        return success;
+    }
+
+    @Override
+    public boolean remove(String productId) {
+
+        //Removes a product from the array, and consequently from the source file as well.
+
+        checkForUpdates();
+        boolean toReturn = false;
+
+
+        for(Product p : productArray){
+            if (p.get(ProductAttribute.UUID).equalsIgnoreCase(productId)){
+                toReturn = productArray.remove(p);
+                break;
+            }
+        }
+
+        updateSource();
+
+        return toReturn;
+    }
+
+    public boolean removeAll(String[] productIds){
+
+        //This function removes all products from the productArray that matches any of the given productIds
+        //It returns whether it was able to find products matching all the id's or not.
+
+        checkForUpdates();
+        int counter = 0;
+
+        for (String productId : productIds) {
+            for (Product p : productArray) {
+
+                if (p.get(ProductAttribute.UUID).equalsIgnoreCase(productId)) {
+                    productArray.remove(p);
+                    counter++;
+                    break;
+                }
+            }
+        }
+
+        return counter == productIds.length;
+    }
+
+    @Override
+    public void setUpdateInterval(int time) {
+        checkForUpdates();
+
+        updateInterval = time;
+    }
+
+    private boolean backgroundUpdate(){
+
+        //This is a background function to be called once every index update interval.
+        //For direct control of index reparses, use reparse()
+
+        if(!backgroundThreadIsRunning) {
+            backgroundThreadIsRunning = true;
+        }
+
+        //Right here is where the XXXX.updateIndex() call to the module from Group 2.2 goes
+
+        return backgroundThread.isAlive();
+    }
+
+    @Override
+    public void run(){
+
+        //This function is only used by the backgroundThread.
+        //It busy-waits (Thread.sleep would work as well) for updateInterval (minutes)
+        //Then reads the json file again and prepares a new ArrayList<Product> for use.
+
+        while(System.currentTimeMillis() < lastCall + (updateInterval * 60000L)){
+            System.out.println();
```

Comment:
Thread.sleep would be a better approach.
Since it release the thread to other processes, and runs their task while your thread sleeps.
This currently starts a thread and blocks it for it's entire life-time.